From 83bf572e9bb5701cfc270319bcdbd4043704282d Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Fri, 2 Feb 2024 12:40:39 +0000
Subject: [PATCH] fixup! vc_mem: Add vc_mem driver for querying firmware memory
 addresses

---
 drivers/char/broadcom/vc_mem.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/char/broadcom/vc_mem.c
+++ b/drivers/char/broadcom/vc_mem.c
@@ -24,6 +24,9 @@
 #include <linux/dma-mapping.h>
 #include <linux/broadcom/vc_mem.h>
 #include <linux/compat.h>
+#include <linux/of.h>
+#include <linux/of_platform.h>
+#include <linux/platform_device.h>
 #include <linux/platform_data/dma-bcm2708.h>
 #include <soc/bcm2835/raspberrypi-firmware.h>
 
