From f8b52bcaec5476312db6c15a797687e6f07bdfff Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Fri, 2 Feb 2024 12:40:58 +0000
Subject: [PATCH] fixup! char: broadcom: Add vcio module

---
 drivers/char/broadcom/vcio.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/char/broadcom/vcio.c
+++ b/drivers/char/broadcom/vcio.c
@@ -19,6 +19,8 @@
 #include <linux/uaccess.h>
 #include <linux/compat.h>
 #include <linux/miscdevice.h>
+#include <linux/of.h>
+#include <linux/platform_device.h>
 #include <soc/bcm2835/raspberrypi-firmware.h>
 
 #define MODULE_NAME "vcio"
