From d473bac69514033c10dbe40f318d33ce9c38ae99 Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Fri, 2 Feb 2024 12:49:56 +0000
Subject: [PATCH] media: i2c: imx258: Squash fixes

---
 drivers/media/i2c/imx258.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/media/i2c/imx258.c
+++ b/drivers/media/i2c/imx258.c
@@ -1012,7 +1012,7 @@ static int imx258_open(struct v4l2_subde
 	try_fmt->field = V4L2_FIELD_NONE;
 
 	/* Initialize try_crop */
-	try_crop = v4l2_subdev_get_try_crop(sd, fh->state, 0);
+	try_crop = v4l2_subdev_state_get_crop(fh->state, 0);
 	try_crop->left = IMX258_PIXEL_ARRAY_LEFT;
 	try_crop->top = IMX258_PIXEL_ARRAY_TOP;
 	try_crop->width = IMX258_PIXEL_ARRAY_WIDTH;
@@ -1307,7 +1307,7 @@ __imx258_get_pad_crop(struct imx258 *imx
 {
 	switch (which) {
 	case V4L2_SUBDEV_FORMAT_TRY:
-		return v4l2_subdev_get_try_crop(&imx258->sd, sd_state, pad);
+		return v4l2_subdev_state_get_crop(sd_state, pad);
 	case V4L2_SUBDEV_FORMAT_ACTIVE:
 		return &imx258->cur_mode->crop;
 	}
