From 2c5634727bc8698114a56c68ba90b81a910dc79d Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Fri, 2 Feb 2024 12:52:12 +0000
Subject: [PATCH] fixup! media: i2c: add ov9281 driver.

---
 drivers/media/i2c/irs1125.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/media/i2c/irs1125.c
+++ b/drivers/media/i2c/irs1125.c
@@ -930,7 +930,7 @@ static int irs1125_detect(struct v4l2_su
 static int irs1125_open(struct v4l2_subdev *sd, struct v4l2_subdev_fh *fh)
 {
 	struct v4l2_mbus_framefmt *format =
-	v4l2_subdev_get_try_format(sd, fh->state, 0);
+	v4l2_subdev_state_get_format(fh->state, 0);
 
 	format->code = MEDIA_BUS_FMT_Y12_1X12;
 	format->width = IRS1125_WINDOW_WIDTH_DEF;
