From ff394f6e46eeee3b8d1ed62e65196f75e91b7490 Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Fri, 2 Feb 2024 12:54:11 +0000
Subject: [PATCH] fixup! media/i2c: Add driver for Omnivision OV2311

---
 drivers/media/i2c/ov2311.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/media/i2c/ov2311.c
+++ b/drivers/media/i2c/ov2311.c
@@ -484,7 +484,7 @@ static int ov2311_set_fmt(struct v4l2_su
 		V4L2_MAP_XFER_FUNC_DEFAULT(fmt->format.colorspace);
 
 	if (fmt->which == V4L2_SUBDEV_FORMAT_TRY) {
-		*v4l2_subdev_get_try_format(sd, sd_state, fmt->pad) =
+		*v4l2_subdev_state_get_format(sd_state, fmt->pad) =
 								fmt->format;
 	} else {
 		ov2311->cur_mode = mode;
@@ -519,7 +519,7 @@ static int ov2311_get_fmt(struct v4l2_su
 
 	mutex_lock(&ov2311->mutex);
 	if (fmt->which == V4L2_SUBDEV_FORMAT_TRY) {
-		fmt->format = *v4l2_subdev_get_try_format(sd, sd_state, fmt->pad);
+		fmt->format = *v4l2_subdev_state_get_format(sd_state, fmt->pad);
 	} else {
 		fmt->format.width = mode->width;
 		fmt->format.height = mode->height;
@@ -596,7 +596,7 @@ __ov2311_get_pad_crop(struct ov2311 *ov2
 {
 	switch (which) {
 	case V4L2_SUBDEV_FORMAT_TRY:
-		return v4l2_subdev_get_try_crop(&ov2311->subdev, sd_state, pad);
+		return v4l2_subdev_state_get_crop(sd_state, pad);
 	case V4L2_SUBDEV_FORMAT_ACTIVE:
 		return &ov2311->cur_mode->crop;
 	}
@@ -802,7 +802,7 @@ static int ov2311_open(struct v4l2_subde
 {
 	struct ov2311 *ov2311 = to_ov2311(sd);
 	struct v4l2_mbus_framefmt *try_fmt =
-				v4l2_subdev_get_try_format(sd, fh->state, 0);
+				v4l2_subdev_state_get_format(fh->state, 0);
 	const struct ov2311_mode *def_mode = &supported_modes[0];
 
 	mutex_lock(&ov2311->mutex);
