From 734cf19609efb1c2863b5775381663c985820062 Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Sun, 27 Mar 2022 13:16:10 +0100
Subject: [PATCH 671/686] hack core to 550

---
 drivers/gpu/drm/vc4/vc4_kms.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/gpu/drm/vc4/vc4_kms.c
+++ b/drivers/gpu/drm/vc4/vc4_kms.c
@@ -945,6 +945,7 @@ vc4_core_clock_atomic_check(struct drm_a
 	}
 
 	hvs_new_state->core_clock_rate = max(cob_rate, pixel_rate);
+	hvs_new_state->core_clock_rate = min(hvs_new_state->core_clock_rate, 550000000);
 
 	return 0;
 }
