From 495efdda520a6fc0c2391a2a770805930031ad52 Mon Sep 17 00:00:00 2001
From: Dom Cobley <popcornmix@gmail.com>
Date: Wed, 6 Apr 2022 13:30:07 +0100
Subject: [PATCH 674/686] fixup! media: i2c: Add driver for IMX519 sensor

---
 drivers/media/i2c/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/media/i2c/Kconfig
+++ b/drivers/media/i2c/Kconfig
@@ -215,7 +215,7 @@ config VIDEO_IMX412
 
 config VIDEO_IMX519
 	tristate "Arducam IMX519 sensor support"
-	depends on I2C && VIDEO_V4L2 && VIDEO_V4L2_SUBDEV_API
+	depends on I2C && VIDEO_DEV && VIDEO_V4L2_SUBDEV_API
 	depends on MEDIA_CAMERA_SUPPORT
 	help
 	  This is a Video4Linux2 sensor driver for the Arducam
